main.ts

```diff
@@ -72,6 +76,44 @@ class Translator {
     this.visit(expr);
   }
 
+  /** Searches for super() calls and emits an initializer for it if found. */
+  maybeEmitInitializerForSuperCall(body: ts.Block) {
+    if (!body || body.statements.length === 0) return;
+
+    var findSuperCalls = (stmt: ts.Statement) => {
+      if (stmt.kind !== ts.SyntaxKind.ExpressionStatement) return;
+      var nestedExpr = (<ts.ExpressionStatement>stmt).expression;
+      if (nestedExpr.kind !== ts.SyntaxKind.CallExpression) return;
+      var callExpr = <ts.CallExpression>nestedExpr;
+      if (callExpr.expression.kind !== ts.SyntaxKind.SuperKeyword) return;
+
+      this.emit(': super (');
+      this.visitList(callExpr.arguments);
+      this.emit(')');
+    };
+
+    body.statements.forEach(findSuperCalls);
+  }
+
+  /**
+   * Checks whether `callExpr` is a super() call that should be ignored because it was already
+   * handled by `maybeEmitSuperInitializer` above.
+   */
+  isHandledSuperCall(callExpr: ts.CallExpression): boolean {
+    if (callExpr.expression.kind !== ts.SyntaxKind.SuperKeyword) return false;
+    // Sanity check that there was indeed a ctor directly above this call.
+    var exprStmt = callExpr.parent;
+    var ctorBody = exprStmt.parent;
+    // TODO(martinprobst): Warn if exprStmt is not the first element of ctorBody.expressions to call
```

Comment:
we'll have to rework our error/warning system at some point anyway, let's do that then.